grails-app/views/report/showRequestDetailReport.gsp

```diff
@@ -0,0 +1,284 @@
+            "error": handleAjaxError,
+            beforeSend : function(){
+              $(".loading").show();
+            },
+            complete: function(){
+              $(".loading").hide();
+            },
+          } );
+        },
+        "oLanguage": {
+          "sZeroRecords": "No records found",
+          "sProcessing": "Loading <img alt='spinner' src='${request.contextPath}/images/spinner.gif' /> Loading... "
+        },
+        "aLengthMenu": [
+          [5, 15, 25, 100, 1000, -1],
+          [5, 15, 25, 100, 1000, "All"]
+        ],
+        "aoColumns":
+          [
+            { "mData": "requestNumber" },
+            { "mData": "dateRequested" },
+            { "mData": "dateIssued" },
+            { "mData": "origin" },
+            { "mData": "destination" },
+            { "mData": "productCode" },
+            { "mData": "productName" },
+            { "mData": "qtyRequested"},
+            { "mData": "qtyIssued"},
+            { "mData": "reasonCodes"},
+            { "mData": "qtyDemand"},
+          ],
+        "dom": '<"top"i>rt<"bottom"flp><"clear">',
+        "aaSorting": [[ 0, "asc" ]],
+      };
+
+      $('#requestDetailReportTable').dataTable(options);
+    }
+
+    function handleAjaxError( xhr, status, error ) {
+      if ( status === 'timeout' ) {
+        alert( 'The server took too long to send the data.' );
+      }
+      else {
+        // User probably refreshed page or clicked on a link, so this isn't really an error
+        if(xhr.readyState == 0 || xhr.status == 0) {
+          return;
+        }
+        alert("An error occurred on the server.  Please contact your system administrator.");
+      }
+    }
+
+    $(document).ready(function() {
+        $(".loading").hide();
+        $('#requestDetailReportTable').dataTable({"bJQueryUI": true});
+        $(".download-button").click(function(event) {
+          event.preventDefault();
+          var params = {
+            "destination.id": $("#destinationId").val(),
+            "origin.id": $("#originId").val(),
+            dateIssuedFrom: $("#dateIssuedFrom").val(),
+            dateIssuedTo: $("#dateIssuedTo").val(),
+            category: $("#category").val(),
+            tags: $("#tags").val(),
+            catalogs: $("#catalogs").val(),
+            product: $("#product").val(),
+            reasonCode: $("#reasonCode").val(),
+            format: "text/csv"
+          };
+          var queryString = $.param(params, true);
+          window.location.href = '${request.contextPath}/json/getRequestDetailReport?' + queryString;
+        });
+
+      $(".submit-button").click(function(event){
+        event.preventDefault();
+        var originId = $("#originId").val();
+        var dateIssuedFrom = $("#dateIssuedFrom").val();
+        var dateIssuedTo = $("#dateIssuedTo").val();
+        var validated = true;
+
+        if (!dateIssuedTo || !dateIssuedFrom || !originId) {
+          alert("All report parameters fields are required");
```

Comment:
How about using $.notify("All report parameters fields are required", "error"); here instead of alerts?